pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,290 @@
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+
+        # Underlying system
+        system = intg.system
+        self._ele_map = system.ele_map
+        self.dtype = intg.system.backend.fpdtype
+
+        # Underlying system elements class
+        self.elementscls = system.elementscls
+
+        # Expressions to plot and configs
+        cons = self.cfg.items_as('constants', float)
+        self._expr = {}
+        self._scene_cfg = self._config_scene(cfgsect, cons)
+        self._pipel_cfg = self._config_pipeline(cfgsect)
+        
+        # Gradient pre-processing
+        self._init_gradients(intg)
+
+        # Base directory for output and counter reset
+        self.basedir = self.cfg.getpath(cfgsect, 'basedir', '.', abs=True)
+        self.nout = {}
+        for sname in self._scene_cfg:
+            name = self._scene_cfg[sname]['image_name']
+            self.nout[name] = file_index(self.basedir, name) if intg.isrestart else 0
+
+        # Set subdivision
+        self.divisor = self.cfg.getint(cfgsect, 'devision', 
+                                       self.cfg.getint('solver', 'order'))
+        self.etypes_div = defaultdict(lambda: self.divisor)
+
+        # Share number of ele types per part
+        ndoms = len(system.ele_map)
+        ndoms = comm.gather(ndoms, root=root)
+        if rank == root:
+            ndoms = np.asarray(ndoms)
+        ndoms = comm.bcast(ndoms, root=root)
+        self.dom_0 = dom_0 = np.sum(ndoms[:rank])
+
+        # Generate conduit blueprint of mesh
+        mesh_n = ConduitNode()
+        self.ops = {}
+        for i, (ename, eles) in enumerate(system.ele_map.items(), start=dom_0):
+            if ename in ['pri', 'pyr']:
+                raise ValueError(f'Ascent: {ename} elements not supported')
+
+            dpf = f'domain_{i}/'
+
+            svpts = self._get_std_ele(ename, eles.nspts)
+            nsvpts = len(svpts)
+
+            mesh_op = self._get_mesh_op(ename, eles.nspts, svpts)
+            soln_op = self._get_soln_op(ename, eles.nspts, svpts)
+            self.ops[ename] = (mesh_op, soln_op)
+
+            mesh_n[dpf + 'state/domain_id'] = rank
+            mesh_n[dpf + 'coordsets/coords/type'] = 'explicit'
+            mesh_n[dpf + 'topologies/mesh/coordset'] = 'coords'
+            mesh_n[dpf + 'topologies/mesh/type'] = 'unstructured'
+            mesh_n[dpf + 'topologies/mesh/elements/shape'] = ename
+
+            subdvcls = subclass_where(BaseShapeSubDiv, name=ename)
+            sconn = subdvcls.subnodes(self.divisor)
+            subdvcon = np.hstack([sconn + j*nsvpts for j in range(eles.neles)])
+            mesh_n[dpf + 'topologies/mesh/elements/connectivity'] = subdvcon
+
+            xd = np.einsum('ik,k...', mesh_op, eles.ploc_at_np('linspts'))
```

Comment:
Where do the linspts come in?  I do not think we use them in the VTK writer?

Reply:
This is the way to get the corner nodes as we previsouly discussed.

Reply:
Just wondering what they are needed for?  As the VTK stuff I don't think uses them.